### PR TITLE
misc: fix injecting --inspect with NODE_OPTIONS

### PR DESCRIPTION
**Pull request by feedthejim**
This PR fixes the passing of the `--inspect` option when calling Next.js with it. It's still not great because you need to target the next file in node_modules directly but I'll add a `next --inspect` option in the future.

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 745af2fac7cb6c2ddabcc21917974f6f109a9fc6

`pnpm test-dev test/development/basic/node-builtins.test.ts` 
- node builtins > should support node.js builtins
- node builtins > should support node.js builtins prefixed by node:

<details>
<summary>Expand output</summary>

  ● node builtins › should support node.js builtins

    expect(received).toInclude(expected)

    Expected string to include:
      "next-router-worker"
    Received:
      "/root/.local/share/fnm/node-versions/v20.8.1/installation/bin/node"

      87 |       expect(parsedData.os).toBe('\n')
      88 |       expect(parsedData.path).toBe('/hello/world/test.txt')
    > 89 |       expect(parsedData.process).toInclude('next-router-worker')
         |                                  ^
      90 |       expect(parsedData.querystring).toBe('a=b')
      91 |       expect(parsedData.stringDecoder).toBe(true)
      92 |       expect(parsedData.sys).toBe(true)

      at Object.toInclude (development/basic/node-builtins.test.ts:89:34)

  ● node builtins › should support node.js builtins prefixed by node:

    expect(received).toInclude(expected)

    Expected string to include:
      "next-router-worker"
    Received:
      "/root/.local/share/fnm/node-versions/v20.8.1/installation/bin/node"

      113 |       expect(parsedData.os).toBe('\n')
      114 |       expect(parsedData.path).toBe('/hello/world/test.txt')
    > 115 |       expect(parsedData.process).toInclude('next-router-worker')
          |                                  ^
      116 |       expect(parsedData.querystring).toBe('a=b')
      117 |       expect(parsedData.stringDecoder).toBe(true)
      118 |       expect(parsedData.sys).toBe(true)

      at Object.toInclude (development/basic/node-builtins.test.ts:115:34)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js feedthejim/fix-inspect | Change |
| - | - | - | - |
| buildDuration | 10.3s | 10.2s | N/A |
| buildDurationCached | 6.1s | 6.1s | N/A |
| nodeModulesSize | 174 MB | 174 MB | N/A |
| nextStartRea..uration (ms) | 531ms | 526ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js feedthejim/fix-inspect | Change |
| - | - | - | - |
| 199-HASH.js gzip | 27.6 kB | 27.6 kB | ✓ |
| 3f784ff6-HASH.js gzip | 53.3 kB | 53.3 kB | ✓ |
| 99.HASH.js gzip | 182 B | 182 B | ✓ |
| framework-HASH.js gzip | 45.3 kB | 45.3 kB | ✓ |
| main-app-HASH.js gzip | 254 B | 252 B | N/A |
| main-HASH.js gzip | 32.9 kB | 32.9 kB | N/A |
| webpack-HASH.js gzip | 1.75 kB | 1.75 kB | N/A |
| Overall change | 126 kB | 126 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js feedthejim/fix-inspect | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js feedthejim/fix-inspect | Change |
| - | - | - | - |
| _app-HASH.js gzip | 206 B | 205 B | N/A |
| _error-HASH.js gzip | 182 B | 180 B | N/A |
| amp-HASH.js gzip | 506 B | 505 B | N/A |
| css-HASH.js gzip | 322 B | 323 B | N/A |
| dynamic-HASH.js gzip | 2.57 kB | 2.57 kB | N/A |
| edge-ssr-HASH.js gzip | 260 B | 259 B | N/A |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 369 B | 369 B | ✓ |
| image-HASH.js gzip | 4.36 kB | 4.36 kB | N/A |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.64 kB | 2.63 kB | N/A |
| routerDirect..HASH.js gzip | 312 B | 311 B | N/A |
| script-HASH.js gzip | 385 B | 384 B | N/A |
| withRouter-HASH.js gzip | 307 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 1.08 kB | 1.08 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js feedthejim/fix-inspect | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 482 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js feedthejim/fix-inspect | Change |
| - | - | - | - |
| index.html gzip | 528 B | 529 B | N/A |
| link.html gzip | 541 B | 542 B | N/A |
| withRouter.html gzip | 523 B | 524 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js feedthejim/fix-inspect | Change |
| - | - | - | - |
| edge-ssr.js gzip | 93.8 kB | 93.8 kB | N/A |
| page.js gzip | 155 kB | 155 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js feedthejim/fix-inspect | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 624 B | 621 B | N/A |
| middleware-r..fest.js gzip | 150 B | 151 B | N/A |
| middleware.js gzip | 22.9 kB | 22.9 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 1.92 kB | 1.92 kB | ✓ |

</details>

</details>
Commit: 745af2fac7cb6c2ddabcc21917974f6f109a9fc6

File: test/development/basic/node-builtins.test.ts [hunk 1]
```diff
@@ -86,7 +86,6 @@ createNextDescribe(
       expect(parsedData.https).toBe(true)
       expect(parsedData.os).toBe('\n')
       expect(parsedData.path).toBe('/hello/world/test.txt')
-      expect(parsedData.process).toInclude('next-router-worker')
```
Comment (ijjk) [inline test/development/basic/node-builtins.test.ts:89]: Why is this removed?

**Review by ijjk — APPROVED** [review]

**Comment by Cow258** [comment]
This PR should fixed #55862 and #56277
